### PR TITLE
bump up pybind11 to support numpy2

### PR DESCRIPTION
**Pull request by MridulS**
Let's see if CI complains.

### CODE REVIEW

**Comment by MridulS** [comment]
@SimonHeybrock I tried to bump pybind and it seems to work if I comment out bits the CI complains about. I'm no pybind expert and I don't have enough context to say if the commented out bits are required with the pybind bump. The CI passes in this PR (but probably not a sufficient condition to see if anything actually breaks)

**Comment by SimonHeybrock** [comment]
Looking at the history, I am not sure why that code exists. I think we did not write this ourselves but used a recipe from somewhere. @jl-wynen do you remember anything?

File: lib/cmake/.conan-recipes/pybind11/conanfile.py [hunk 1]
```diff
@@ -65,18 +65,18 @@ def package(self):
...
+            #     os.path.join(
+            #         self.package_folder,
+            #         "lib",
+            #         "cmake",
+            #         "pybind11",
+            #         "pybind11Common.cmake",
+            #     ),
+            #     "if(TARGET pybind11::lto)",
+            #     "if(FALSE)",
+            # )
```
Comment (SimonHeybrock) [inline lib/cmake/.conan-recipes/pybind11/conanfile.py:79]: Remove all the commented code here and below?

**Review by jl-wynen — APPROVED** [review]

**Comment by MridulS** [comment]
I'll go ahead and merge this one. We can see if anything messes up with the nightly build :D

**Comment by MridulS** [comment]
Nightly build passed just fine for scipp release https://github.com/scipp/scipp/actions/runs/8929953448

**Pull request merged**